### PR TITLE
server/api: improve coverage

### PR DESCRIPTION
**Pull request by nolouch**
PTAL @siddontang @huachaohuang @overvenus

### CODE REVIEW

File: server/api/label_test.go [hunk 1]
```diff
@@ -86,32 +100,61 @@ func (s *testLabelsStoreSuite) TestStroesLabelFilter(c *C) {
...
+		mustPutStore(c, s.svr, store)
+	}
+	s.client = newUnixSocketClient()
+}
+
+func (s *testLabelsStoreSuite) TearDownSuite(c *C) {
+	s.cleanup()
+}
+
+func (s *testLabelsStoreSuite) TestStroesLabelGet(c *C) {
```
Comment (siddontang) [inline server/api/label_test.go]: TestStoresLabelGet

**Comment by siddontang** [comment]
Please fix conflict.

File: server/api/label_test.go [hunk 2]
```diff
@@ -120,9 +163,13 @@ func (s *testLabelsStoreSuite) TestStroesLabelFilter(c *C) {
 		},
 	}
 	for _, t := range table {
-		f, err := newStoresLabelFilter(t.name, t.value)
+		resp, err := s.client.Get(fmt.Sprintf("%s/labels/stores?name=%s&value=%s", s.urlPrefix, t.name, t.value))
+		defer resp.Body.Close()
```
Comment (siddontang) [inline server/api/label_test.go]: don't call defer in for loop. call Close directly after readJSON.

File: server/api/store_test.go [hunk 3]
```diff
@@ -14,54 +14,164 @@
...
+			mapWant[s.Id] = s
+		}
+	}
+	for _, s := range ss {
+		c.Assert(s.Store.Store, DeepEquals, mapWant[s.Store.Store.Id])
+	}
+}
+
+func (s *testStoreSuite) TestStoresList(c *C) {
+	resp, err := s.client.Get(fmt.Sprintf("%s/stores", s.urlPrefix))
```
Comment (siddontang) [inline server/api/store_test.go]: can you extract a function to get the json body and decode to an interface, so many test cases can use it.

File: server/api/store_test.go [hunk 4]
```diff
@@ -14,54 +14,164 @@
...
+		{
+			id:     7,
+			status: http.StatusInternalServerError,
+		},
+	}
+	for _, t := range table {
+		req, err := http.NewRequest(http.MethodDelete, fmt.Sprintf("%s/store/%d", s.urlPrefix, t.id), nil)
+		c.Assert(err, IsNil)
+		resp, err := s.client.Do(req)
+		defer resp.Body.Close()
```
Comment (siddontang) [inline server/api/store_test.go]: ditto

File: server/api/server_test.go [hunk 5]
```diff
@@ -117,3 +142,51 @@ func mustWaitLeader(c *C, svrs []*server.Server) *server.Server {
...
+	mustRPCRequest(c, s.GetAddr(), req)
+}
+
+func readJSONWithURL(url string, data interface{}) error {
+	resp, err := unixClient.Get(url)
+	defer resp.Body.Close()
+	if err != nil {
+		return err
+	}
+	err = readJSON(resp.Body, data)
```
Comment (siddontang) [inline server/api/server_test.go]: return err or readJSON directly

**Comment by siddontang** [comment]
PTAL @overvenus

File: server/api/server_test.go [hunk 6]
```diff
@@ -117,3 +142,47 @@ func mustWaitLeader(c *C, svrs []*server.Server) *server.Server {
 	c.Fatal("no leader")
 	return nil
 }
+
+func mustRPCRequest(c *C, addr string, request *pdpb.Request) *pdpb.Response {
```
Comment (overvenus) [inline server/api/server_test.go]: Why not use https://github.com/pingcap/pd/blob/c77e51541f7c9df54af3586d13d377240e4f9a01/pkg/testutil/testutil.go#L36

**Comment by siddontang** [comment]
why coverage still decreases?

**Comment by nolouch** [comment]
PTAL @siddontang

**Comment by siddontang** [comment]
LGTM

**Pull request merged**